app/questionnaire/questionnaire_store_updater.py

```diff
@@ -339,43 +387,86 @@ def update_answers(
 
+        if answers_updated:
+            self.capture_progress_section_dependencies()
+
+    def capture_progress_section_dependencies(self) -> None:
+        self._capture_section_dependencies_progress_value_source_for_current_block()
+        self._capture_section_dependencies_progress_value_source_for_current_section()
+
     def update_progress_for_dependent_sections(self) -> None:
         """Removes dependent blocks from the progress store and updates the progress to IN_PROGRESS.
         Section progress is not updated for the current location as it is handled by `handle_post` on block handlers.
         """
+        evaluated_dependents: list[tuple] = []
 
         self._remove_dependent_blocks_and_capture_dependent_sections()
 
-        for section in self.dependent_sections:
+        chronological_dependents = self.get_chronological_section_dependents()
+
+        for section in chronological_dependents:
             if (
                 section.section_id,
                 section.list_item_id,
             ) not in self.started_section_keys():
                 continue
 
-            is_path_complete = section.is_complete
-            if is_path_complete is None:
-                is_path_complete = self._router.is_path_complete(
-                    self._router.routing_path(
-                        section.section_id, list_item_id=section.list_item_id
-                    )
+            if (section.section_id, section.list_item_id) not in evaluated_dependents:
+                self.evaluate_dependents(
+                    section.section_id,
+                    section.list_item_id,
+                    section.is_complete,
+                    evaluated_dependents,
                 )
 
-            self.update_section_status(
-                is_complete=is_path_complete,
-                section_id=section.section_id,
-                list_item_id=section.list_item_id,
+    def evaluate_dependents(
+        self,
+        section_id: str,
+        list_item_id: str | None,
+        is_complete: bool | None,
+        evaluated_dependents: list,
+    ) -> None:
+        is_path_complete = is_complete
+        if is_path_complete is None:
+            is_path_complete = self._router.is_path_complete(
+                self._router.routing_path(section_id, list_item_id=list_item_id)
             )
 
+        if self.update_section_status(
+            is_complete=is_path_complete,
+            section_id=section_id,
+            list_item_id=list_item_id,
+        ):
+            dependents_of_dependent = self.get_dependents_of_section(section_id)
+            for dependent in dependents_of_dependent:
+                if repeating_list := self._schema.get_repeating_list_for_section(
+                    dependent
+                ):
+                    for item_id in self._list_store[repeating_list].items:
+                        if (section_id, item_id) not in evaluated_dependents:
```

Comment:
This should be the current lowest level dependent you are looping over
```suggestion
                        if (dependent, item_id) not in evaluated_dependents:
```

Worth renaming to `dependent_section_id` for clarity.

Reply:
Updated 👍